notebooks/1.data_splits/nbconverted/1.1_cell_injury_data_splits.py

```diff
@@ -0,0 +1,580 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Splitting feature selected cell injury profiles
+#
+# Here, we utilize the feature-selected cell-injury profiles generated in the preceding module notebook [here](../0.feature_selection/0.feature_selection.ipynb), focusing on dividing the data into training, testing, and holdout sets for machine learning training.
+#
+
+# In[1]:
+
+
+import json
+import pathlib
+import sys
+import warnings
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+sys.path.append("../../")  # noqa
+from src.utils import (  # noqa
+    get_injury_treatment_info,
+    load_json_file,
+    split_meta_and_features,
+)
+
+# ignoring warnings
+warnings.catch_warnings(action="ignore")
+
+
+# ## Parameters
+#
+# Below are the parameters defined that are used in this notebook
+#
+
+# In[2]:
+
+
+# setting seed constants
+seed = 0
+np.random.seed(seed)
+
+# directory to get all the inputs for this notebook
+data_dir = pathlib.Path("../../data").resolve(strict=True)
+results_dir = pathlib.Path("../../results").resolve(strict=True)
+fs_dir = (results_dir / "0.feature_selection").resolve(strict=True)
+
+# directory to store all the output of this notebook
+data_split_dir = (results_dir / "1.data_splits").resolve()
+data_split_dir.mkdir(exist_ok=True)
+
+
+# In[3]:
+
+
+# loading feature selected cell injury dataset
+fs_profile_path = (fs_dir / "fs_cell_injury_only.csv.gz").resolve(strict=True)
+
+# load in feature select cell injury feature space file
+fs_feature_space = load_json_file(
+    fpath=(fs_dir / "fs_cell_injury_only_feature_space.json").resolve(strict=True)
+)
+fs_feats = fs_feature_space["features"]
+
+# load data
+fs_profile_df = pd.read_csv(fs_profile_path)
+
+
+# splitting meta and feature column names
+fs_meta, _ = split_meta_and_features(fs_profile_df)
+
+# display
+print("fs profile with control: ", fs_profile_df.shape)
+fs_profile_df.head()
+
+
+# ## Exploring the data set
+#
+# Below is a exploration of the selected features dataset. The aim is to identify treatments, extract metadata, and gain a understanding of the experiment's design.
+#
+
+# Below demonstrates the amount of wells does each treatment have.
+#
+
+# In[4]:
+
+
+# displaying the amount of wells per treatments
+well_treatments_counts_df = (
+    fs_profile_df["Compound Name"].value_counts().to_frame().reset_index()
+)
+
+well_treatments_counts_df
+
+
+# Below we show the amount of wells does a specific cell celluar injury has
```

Comment:
need to finish sentence - oh, but maybe just delete "does"?

Reply:
Fixed!